src/python/pants/backend/python/subsystems/lambdex.lock

```diff
@@ -54,28 +54,28 @@
           "artifacts": [
             {
               "algorithm": "sha256",
-              "hash": "bd3a8c500d11b439d0a42b3b117413a1e133d0ee1b6b2c61ec6e44e86b53fe6f",
-              "url": "https://files.pythonhosted.org/packages/40/90/b260cbedb249fd24e733d524f7d6f85052b45f07c5e6cf0c7bc28058c4e4/pex-2.1.135-py2.py3-none-any.whl"
+              "hash": "5031c3b283d63470faeaf82d0fc1344e6f71b3ad4ac4ca34572a42bae6dfc4b8",
+              "url": "https://files.pythonhosted.org/packages/e8/6e/eadca769b580a93d10caeca29d17397565672cf8b675991ccbf959c75476/pex-2.1.137-py2.py3-none-any.whl"
             },
             {
               "algorithm": "sha256",
-              "hash": "87a9eff7522423eebe70b1d2f026309bdb5d75b8e238e5ac764d7ed7b3eeb6f5",
-              "url": "https://files.pythonhosted.org/packages/35/ee/af772c967c288e3175a9b37e226543778c7215dc06da2c920b9c5b91123b/pex-2.1.135.tar.gz"
+              "hash": "cb0ce6cf64757dd5ba4f34c4607ab485f7909e6c24cd479ca28ce52205f0edeb",
+              "url": "https://files.pythonhosted.org/packages/a8/06/26c731fbf11fad3b1dff7b1d535636c65d8d630eabc981ca025d2e7b5cfb/pex-2.1.137.tar.gz"
             }
           ],
           "project_name": "pex",
           "requires_dists": [
             "subprocess32>=3.2.7; extra == \"subprocess\" and python_version < \"3\""
           ],
           "requires_python": "!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,<3.12,>=2.7",
-          "version": "2.1.135"
+          "version": "2.1.137"
         }
       ],
       "platform_tag": null
     }
   ],
   "path_mappings": {},
-  "pex_version": "2.1.135",
+  "pex_version": "2.1.137",
   "pip_version": "23.0.1",
```

Comment:
@benjyw is there a reason `--pip-version` is pinned low for tool locks? If not I can change to `latest`. Basically the fact the user reqs are locked differently than the tool reqs is what I'm chasing down.

Reply:
No special reason, just conservativeness. I think we should default to `latest` for tool and user lockfiles - any differences between resolves due to different pip versions will likely be dwarfed by differences due to changes on PyPI.

Reply:
Ok, thanks. Done.